### DISCUSSION TITLE
将源码和文档编写在同一个文件中，即dtx文件

### DISCUSSION THREAD

**Discussion by aqni** [Tr0py/NKU-thesis-template-2020#33] [Ideas]
我发现直接编写sty文件或cls文件的模板较少，大部分模板将模板和文档一起编写在一个dtx文件中（类似于c语言中常见的内嵌文档Doxygen），例如：

-  [pkuthss](https://github.com/CasperVector/pkuthss)
-  [thuthesis](https://github.com/tuna/thuthesis)
-  [fduthesis](https://github.com/stone-zeng/fduthesis)

一些参考资料：
- https://texfaq.org/FAQ-dtx
- http://mirrors.ctan.org/info/dtxtut/dtxtut.pdf
- http://mirrors.ctan.org/macros/latex/base/ltxdoc.pdf
- http://mirrors.ctan.org/macros/latex/base/docstrip.pdf
- https://tex.stackexchange.com/questions/25116/what-is-good-practice-when-preparing-a-package-for-ctan

我先完成内容上的重构和校对，有时间再这样的完成项目结构上的重构。

**Comment by Tr0py** [comment]
按照我的理解，dtx的主要的好处是可以用tex来把文档以注释的方式整合到源码里，之后再次被拆分sty+doc。

我个人不是很喜欢dtx因为在注释里写一堆tex会让代码看起来很乱。你现在的`.cls`读起来精简易懂，用markdown写的doc读起来精简易懂。但如果是把markdown的doc变成tex加到`.cls`注释中可能会让人读源码时望而生畏。dtx的主要意图我的理解是在早些时候很方便地传输一个package本身，然后由dev和user自己编译doc。这个使用场景应该不太符合我们现在：dev希望有简洁的源码，user希望有简洁明了的现成的doc，并不想自己编译doc。

所以我其实觉得如果我们不发布单独package的话，当前的文档与源码分离的模式更适合开发和使用。

但不管怎样，这只是我的看法。如果你有兴趣的话，当然可以做这个改动让模板更符合社区的规范hh

**Reply by aqni** [reply]
我先开个分支尝试一下，看看效果

**Reply by aqni** [reply]
我的想法是：

- dtx主要写文档类的接口和实现，因为代码实现经常需要改动，单独写文档的话，可能造成文档更新不及时。

- FQA，安装，使用示例仍然用单独的文档写。

**Reply by Tr0py** [reply]
ok，听起来那也可以。我最初的考虑是尽量为后续开发减少难度。
